jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/DynamicUtils.java

```
@@ -55,195 +55,262 @@
+                    return executeSubProcess((StatefulKnowledgeSessionImpl) ksession,
+                                             processId,
+                                             parameters,
+                                             processInstance,
+                                             subProcessNodeInstance);
                 }
             });
-    	} else {
-    		throw new IllegalArgumentException("Unsupported ksession: " + ksession == null ? "null" : ksession.getClass().getName());
-    	}
-	}
-	
-	private static long executeSubProcess(StatefulKnowledgeSessionImpl ksession, String processId, Map<String, Object> parameters, ProcessInstance processInstance, SubProcessNodeInstance subProcessNodeInstance) {
-		Process process = ksession.getKieBase().getProcess(processId);
+        } else {
+            throw new IllegalArgumentException("Unsupported ksession: " + ksession == null ? "null" : ksession.getClass().getName());
+        }
+    }
+
+    private static long executeSubProcess(StatefulKnowledgeSessionImpl ksession,
+                                          String processId,
+                                          Map<String, Object> parameters,
+                                          ProcessInstance processInstance,
+                                          SubProcessNodeInstance subProcessNodeInstance) {
+        Process process = ksession.getKieBase().getProcess(processId);
         if (process == null) {
-            logger.error("Could not find process {}", processId);
+            logger.error("Could not find process {}",
+                         processId);
             logger.error("Aborting process");
-        	processInstance.setState(ProcessInstance.STATE_ABORTED);
-        	
-        	return -1;
+            processInstance.setState(ProcessInstance.STATE_ABORTED);
+
+            return -1;
```

I would prefer to throw an exeption here instead of returning -1.